README.md

```diff
@@ -1,24 +1,648 @@
+- Redis 7+
+- FFmpeg with CUDA support
+- NVIDIA GPU with NVENC/NVDEC support
+- NVIDIA CUDA Toolkit 12.0.1
 
-* Deployment instructions
+**Optional:**
+- Docker & Docker Compose
+- asdf version manager
 
-* ...
+### Quick Setup
+
+1. **Clone the repository**
+
+   ```bash
+   git clone <repository-url>
+   cd media
+   ```
+
+2. **Install dependencies**
+
+   ```bash
+   # Using asdf for version management
+   asdf install
+
+   # Install Ruby gems
+   bundle install
+   ```
+
+3. **Setup database**
+
+   ```bash
+   # Start PostgreSQL and Redis
+   docker-compose up -d
+
+   # Create and setup database
+   bin/rails db:create
+   bin/rails db:schema:load
+   bin/rails db:data:migrate  # Load transcoding profiles
+   ```
+
+4. **Configure environment**
+
+   Create `config/settings/development.local.yml`:
+   ```yaml
+   jwt_secret: your_development_secret
+   ```
+
+5. **Start the application**
+
+   ```bash
+   # Start web server and background workers
+   foreman start
+
+   # Or individually:
+   # bin/rails server -p 3009
+   # bundle exec sidekiq -q low -q default -q high -q critical -q gpu
+   ```
+
+6. **Access the application**
+   - API: http://localhost:3009
+   - Sidekiq UI: http://localhost:3009/sidekiq
+
+---
+
+## Key Features
+
+### Core Capabilities
+- **GPU-Accelerated Transcoding** - NVIDIA CUDA hardware encoding/decoding with NVENC/NVDEC
+- **Multiple Quality Profiles** - Automatic transcoding to 480p, 720p, and 1080p
+- **AV1 Encoding** - Modern AV1 codec with hardware acceleration (av1_nvenc)
+- **Asynchronous Processing** - Background job processing with Sidekiq
+- **Quality Validation** - Automatic resolution detection and upscaling prevention
+- **Cloud Storage** - S3 and IBM Cloud Object Storage integration
+- **RESTful API** - JWT-authenticated API endpoints
+- **Concurrent Processing** - Multi-queue job management with dedicated GPU queue
+
+### Supported Formats
+
+**Input Video Formats:**
+- MP4 (video/mp4)
+- WebM (video/webm)
+- QuickTime/MOV (video/quicktime)
+
+**Output Specifications:**
+| Quality | Resolution | Codec | Video Bitrate | Audio |
+|---------|-----------|-------|---------------|-------|
+| 480p | 854×480 | av1_nvenc | 1000k | AAC 128k |
+| 720p | 1280×720 | av1_nvenc | 1800k | AAC 128k |
+| 1080p | 1920×1080 | av1_nvenc | 2900k | AAC 128k |
+
+---
+
+## Technology Stack
+
+| Component | Technology |
+|-----------|-----------|
+| **Framework** | Ruby on Rails 8.1.0 (API-only) |
+| **Language** | Ruby 3.4.7 |
+| **Database** | PostgreSQL 16 with pgvector |
+| **Cache/Queue** | Redis 7 |
+| **Job Processing** | Sidekiq 8.x with sidekiq-cron |
+| **Web Server** | Puma |
+| **Video Processing** | FFmpeg with CUDA 12.0.1 |
+| **GPU Acceleration** | NVIDIA NVENC, NVDEC, CUVID, LibNPP |
+| **Storage** | Active Storage (S3/IBM COS) |
+| **Authentication** | JWT |
+| **Testing** | RSpec 6.0+, FactoryBot |
+| **Monitoring** | Instana APM |
+| **CI/CD** | GitHub Actions |
+
+---
+
+## API Documentation
```

Comment:
api docs will be very difficult to maintain this way, my recommendation would be to remove this and instead use swagger with rswag
https://github.com/rswag/rswag